File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        int[] atoz = new int[26];
        
        if (s.length() != t.length()) return false;
        
        for (int i = 0; i < s.length(); i++) {
            atoz[s.charAt(i) - 'a']++;
            atoz[t.charAt(i) - 'a']--;
        }
        
        for (int i : atoz) {
            if (i != 0) return false;
        }
        
        return true;
    }
}

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        
        smap = {}
        tmap = {}
        
        for i in range(len(s)):
            char = s[i]
            smap[char] = smap.get(char, 0) + 1
        
        for i in range(len(t)):
            char = t[i]
            tmap[char] = tmap.get(char, 0) + 1
        
        for key in smap:
            if key not in tmap or tmap[key] != smap[key]:
                return False

        
        return True
        